/* Arquivo de definição da interface entre cliente e servidor
 * Estamos usando a versão 3 de protocol buffers 
 */
syntax = "proto3";

// Nome do pacote 
package hello;

service DoStuff {
  /* Estamos definindo dois métdos; em ambos o parâmetro
   *   é do tipo HelloRequest e a resposta é do tipo HelloReply.
   */
  rpc say_hello (HelloRequest) returns (HelloReply) {}
  rpc say_hello_again (HelloRequest) returns (HelloReply) {}
}

// HelloRequest tem apenas um campo, que é o pid do cliente
message HelloRequest {
  int32 pid = 1;  // Cada campo deve ser numerado
}

// A resposta também só tem um componente, um string
message HelloReply {
  string retval = 1;
}
